execution_engine_testing/test_support/src/auction.rs

```diff
@@ -0,0 +1,402 @@
+
+        let txn = lmdb_env.begin_ro_txn().unwrap();
+        let mut cursor = txn.open_ro_cursor(db).unwrap();
+
+        let existing_keys = cursor
+            .iter()
+            .map(|(key, _)| Digest::try_from(&*key).expect("should be a digest"));
+        necessary_tries.extend(existing_keys);
+    }
+    writeln!(
+        report_writer,
+        "height,db-size,transfers,time_ms,necessary_tries,total_tries"
+    )
+    .unwrap();
+    // simulating a block boundary here.
+    for current_block in 0..block_count {
+        let start = Instant::now();
+        total_transfers += exec_requests.len();
+
+        transfer::transfer_to_account_multiple_native_transfers(
+            &mut builder,
+            &exec_requests,
+            use_scratch,
+        );
+        let transfer_root = builder.get_post_state_hash();
+        let maybe_auction_root = if run_auction {
+            if use_scratch {
+                step_and_run_auction(&mut builder, &validator_keys);
+            } else {
+                builder.advance_era(
+                    validator_keys
+                        .iter()
+                        .cloned()
+                        .map(|id| RewardItem::new(id, 1)),
+                );
+                builder.commit();
+            }
+            Some(builder.get_post_state_hash())
+        } else {
+            None
+        };
+        let exec_time = start.elapsed();
+        find_necessary_tries(
+            builder.get_engine_state(),
+            &mut necessary_tries,
+            transfer_root,
+        );
+
+        if let Some(auction_root) = maybe_auction_root {
+            find_necessary_tries(
+                builder.get_engine_state(),
+                &mut necessary_tries,
+                auction_root,
+            );
+        }
+
+        let total_tries = {
+            let engine_state = builder.get_engine_state();
+            let lmdb_env = engine_state.get_state().environment().env();
+            let db = engine_state.get_state().trie_store().get_db();
+            let txn = lmdb_env.begin_ro_txn().unwrap();
+            let mut cursor = txn.open_ro_cursor(db).unwrap();
+            cursor.iter().count()
+        };
+
+        if use_scratch {
+            // This assertion is only valid with the scratch trie.
+            assert_eq!(
+                necessary_tries.len(),
+                total_tries,
+                "should not create unnecessary tries"
+            );
+        }
+
+        writeln!(
+            report_writer,
+            "{},{},{},{},{},{}",
+            current_block,
+            builder.lmdb_on_disk_size().unwrap(),
+            total_transfers,
+            exec_time.as_millis() as usize,
+            necessary_tries.len(),
+            total_tries,
+        )
+        .unwrap();
+        report_writer.flush().unwrap();
+    }
+}
+
+// find all necessary tries - hoist to FN
+fn find_necessary_tries<S>(
```

Comment:
This method is basically an iterator over all tries under `state_root`, correct? It _could_ be a method in the `trie.rs` itself (or some other helper file).